### ISSUE TITLE
Update serve-index to resolve a vulnerability

### ISSUE THREAD

**Issue by Fydon** [jhipster/generator-jhipster#8132]
##### **Overview of the issue**

NPM reports multiple vulnerability, where updating serve-index with the following command will resolve 1 vulnerability: `npm update serve-index --depth 2`

Low             Regular Expression Denial of Service
Package         debug
Dependency of   webpack-dev-server [dev]
Path            webpack-dev-server > serve-index > debug
More info       https://nodesecurity.io/advisories/534

##### **Motivation for or Use Case**

Eliminate vulnerabilities

##### **Reproduce the error**

Create a new react or angular project

##### **Related issues**

Didn't find one after searching

##### **Suggest a Fix**

I'm not sure of a fix for npm, other than someone manually running the command above or adding another dependency. For yarn the fix appears to be to add the following to [Angular's package.json.ejs][1] and [React's package.json.ejs][2]:

      "resolutions": {
      "**/debug": "^2.6.9 || >=3.1"
    },

##### **JHipster Version(s)**

```
XXX@0.0.0 XXX
`-- generator-jhipster@5.2.1

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "XXX",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.2.1",
    "applicationType": "monolith",
    "baseName": "XXX",
    "packageName": "XXX",
    "packageFolder": "XXX",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "oracle",
    "prodDatabaseType": "oracle",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

git version 2.17.1.windows.2

node: v8.11.4

npm: 6.4.0

yeoman: 2.0.5

Docker version 18.06.0-ce, build 0ffa825

docker-compose version 1.22.0, build f46880fe

- [X] Checking this box is mandatory (this is just to show you read everything)

[1]: https://github.com/jhipster/generator-jhipster/blob/1a037d2735b2dcdd6c7bff2b7510fa4257e0451a/generators/client/templates/angular/package.json.ejs
[2]: https://github.com/jhipster/generator-jhipster/blob/395f7ffcad1081c4f6faa2f9cf28affadb25700f/generators/client/templates/react/package.json.ejs

**Comment by jdubois** [comment]
I totally understand this makes the generated project look bad, but then if it's a dependency on webpack-dev-server there shouldn't be any issue at all once the project is build... Shouldn't this be fixed with a new release of webpack-dev-server?

**Comment by jdubois** [comment]
I'm upgrading webpack-dev-server as it seems to solve the issue
@Fydon how do you scan for those vulnerabilities? I'm sorry but I can't reproduce, so I can't test - I'm just reading the `yarn.lock` file to try to understand if it's fixed

**Comment by jdubois** [comment]
So my upgrade seems to do the trick, and this is why I'll close this with my commit, but @Fydon if you can test it's most welcome, as I can't reproduce it like you

**Comment by Fydon** [comment]
@jdubois Good point about this being dev only. Come to think of it all the issues I raised were for development dependencies, which is why Sonar and Github haven't alerted about them.

`npm install` informed me of the issue and then I used [npm audit](https://docs.npmjs.com/cli/audit) to provide the details. npm is good in that it tells you how to solve the problem in addition to advising of the problem, where this is only a manual fix listed by npm. The latest version of webpack-dev-server still has a dependency on an older version of debug, although the resolutions section should fix the issue when using yarn.

I was using npm as I experienced problems with yarn on another PC, but have now switched to yarn so that I can use resolutions in the future to fix vulnerabilities in production builds.
